### PR TITLE
Update Azure Redis resourceclass YAML to v1beta1

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

Updates Azure Redis YAMLs and API reference for v1beta1 bump.

Part of #863
Depends on https://github.com/crossplaneio/stack-azure/pull/58
Fixes https://github.com/crossplaneio/stack-azure/issues/50 

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation], [examples], or [release notes].
- [x] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

[documentation]: https://github.com/crossplaneio/crossplane/tree/master/docs
[examples]: https://github.com/crossplaneio/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplaneio/crossplane/tree/master/PendingReleaseNotes.md
[`clusterrole.yaml`]: https://github.com/crossplaneio/crossplane/blob/master/cluster/charts/crossplane/templates/clusterrole.yaml

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -84,7 +84,7 @@ following API groups:
 API Group | Status | Details
 --------  | ------ | -------
 [`azure.crossplane.io/v1alpha3`] | Alpha | Core Azure resources.
-[`cache.azure.crossplane.io/v1alpha3`] | Alpha | Azure Redis managed resources and classes.
+[`cache.azure.crossplane.io/v1beta1`] | Beta | Azure Redis managed resources and classes.
```
Comment (hasheddan) [inline docs/api.md]: This link will be broken until the API docs are regenerated. Do you think we should go ahead and regenerate the docs for just Azure Redis here, or just wait until the next release and regenerate for all updated resources?

Comment (negz) [reply docs/api.md]: Probably worth regenerating the API docs as part of this PR. https://github.com/negz/crossdocs/pull/1 shows how @jbw976 bumped it for 0.4. It's probably much less work for this change - I'm guessing just replacing cache v1alpha3 with v1beta and running crossdocs.

Comment (jbw976) [reply docs/api.md]: lemme know if you run into any issues @hasheddan, i should still remember most of my recent experience in that crossdocs tool 🤔

Comment (muvaf) [reply docs/api.md]: Done!

**Pull request merged**